network/p2p/middleware.go

```diff
@@ -58,7 +58,17 @@ const (
 	LargeMsgUnicastTimeout = 1000 * time.Second
 )
 
-var _ network.Middleware = (*Middleware)(nil)
+var (
+	_ network.Middleware = (*Middleware)(nil)
+
+	// unauthenticated is the default callback func used to check if a node
+	// is staked or not in the topic validator before decoding network messages.
+	// This default func should only be used on public network channels, since all
+	// nodes can participate on public channels this func always returns true.
```

Comment:
let's make this warning louder
```suggestion
	// unauthenticated should ONLY be used for public network channels to allow
	// messages from any peer. This is NOT safe for channels on the staked network
	// as it skips authenticating the peer which may result in resource exhaustion attacks.
```